index.js

```diff
@@ -313,4 +386,95 @@ function reducer(acc, cur) {
   return acc;
 }
 
+/**
+ * Given an error from got, see if Shopify told us how long to wait before
+ * retrying. Return a duration in ms if we can, and otherwise return null.
+ *
+ * @param {Object} error Error object from got call
+ * @return {Boolean | null}
+ * @private
+ **/
+function maybeRetryMS(error) {
+  // for simplicity, retry network connectivity issues after a hardcoded 1s
+  if (
+    typeof error === 'object' &&
+    'code' in error &&
+    RetryableErrorCodes.has(error.code)
+  ) {
+    return 1000;
+  }
+
+  const response = error.response;
+  if (!response || !response.headers) {
+    return null;
+  }
+
+  if (response.headers && response.headers['retry-after']) {
+    const value = parseFloat(response.headers['retry-after']);
+    if (isNaN(value)) {
+      const when = new Date(value).valueOf();
+      return when - Date.now();
```

Comment:
This still returns `NaN` because `Number.isNaN(new Date(NaN).valueOf()) === true.`

Reply:
Ah I messed this up. I will fix it -- it's supposed to date parse the It was actually supposed to be an attempt to parse timestamps (not durations) returned in the `Retry-After` header, which was a thing we observed Shopify do at one point. They don't document that they do that anywhere so I can remove this if you wish, but we definitely saw them do it at some point.